src/main/java/wolox/training/web/controllers/BookController.java

```diff
@@ -0,0 +1,109 @@
+package wolox.training.web.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+import wolox.training.web.dtos.BookCreationRequestDto;
+
+/**
+ * The {@link Book}s REST controller.
+ */
+@RestController
+@RequestMapping(value = "books", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+public class BookController {
+
+    /**
+     * The {@link BookRepository} used to read and write {@code Book} data.
+     */
+    private final BookRepository bookRepository;
+
+
+    /**
+     * Constructor.
+     *
+     * @param bookRepository The {@link BookRepository} used to read and write {@link Book} data.
+     */
+    @Autowired
+    public BookController(final BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+
+    /**
+     * Endpoint for getting all the {@link Book}s in the system.
+     *
+     * @return A {@link ResponseEntity} of {@link Iterable} of {@link Book}s stored in the system.
+     * Might be empty.
+     */
+    @GetMapping
+    public ResponseEntity<Iterable<Book>> getAllBooks() {
+        final var books = bookRepository.findAll();
+        return ResponseEntity.ok(books);
+    }
+
+    /**
+     * Endpoint for getting a {@link Book} by its {@code id}.
+     *
+     * @param id The id of the {@link Book} to be retrieved.
+     * @return A {@link ResponseEntity} containing the {@link Book} with the given {@code id} if it
+     * exists, or with 404 Not Found {@link ResponseEntity} otherwise.
+     */
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> getById(@PathVariable("id") final long id) {
+        final var book = bookRepository.findById(id);
```

Comment:
is the a Java 10 var? or something else?

Reply:
Yes, Java 10 var. I'm using Java 11 though